core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -306,7 +306,7 @@
     /**
      * @return whether connector handles missing columns during insert
      */
-    boolean supportsMissingColumnsOnInsert(Session session, TableHandle tableHandle);
+    boolean supportsMissingColumnsOnInsert(Session session, TableHandle tableHandle, QualifiedObjectName qualifiedObjectName);
```

Comment:
Adding a test should not modify the existing product code? `TableHandle` should already have information about its name, so why you are passing `qualifiedObjectName`?